From: Ping Cheng <pinglinux@gmail.com>
Date: Sun, 24 Jun 2012 23:44:46 -0500
Subject: wacom: ignore new-style Wacom multi touch packets on MT Tablet PC
Bug-Debian: http://bugs.debian.org/677164

Tablets such as 0xE6 (Thinkpad x220t) already worked fine before
adding support for the new packet format, so let's drop the
functionality for such devices for now.  Meanwhile 0xE5 can still use
the new packet format.

This should bring the behavior of TABLETPC2FG devices closer to that
from before 1963518b9b1b (Input: wacom - add 0xE5 (MT device) support,
2012-04-29).

[jn: extracted from a larger commit in the input-wacom repository,
 with new description]

Signed-off-by: Ping Cheng <pingc@wacom.com>
Signed-off-by: Jonathan Nieder <jrnieder@gmail.com>
---
 drivers/input/tablet/wacom_wac.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/input/tablet/wacom_wac.c b/drivers/input/tablet/wacom_wac.c
index 004bc1bb1544..d696ab7ecc2b 100644
--- a/drivers/input/tablet/wacom_wac.c
+++ b/drivers/input/tablet/wacom_wac.c
@@ -1547,7 +1547,6 @@ int wacom_setup_input_capabilities(struct input_dev *input_dev,
 		__set_bit(INPUT_PROP_POINTER, input_dev->propbit);
 		break;
 
-	case TABLETPC2FG:
 	case MTSCREEN:
 		if (features->device_type == BTN_TOOL_FINGER) {
 
@@ -1559,6 +1558,11 @@ int wacom_setup_input_capabilities(struct input_dev *input_dev,
 
 			for (i = 0; i < features->touch_max; i++)
 				wacom_wac->slots[i] = -1;
+		}
+		/* fall through */
+
+	case TABLETPC2FG:
+		if (features->device_type == BTN_TOOL_FINGER) {
 
 			input_mt_init_slots(input_dev, features->touch_max);
 			input_set_abs_params(input_dev, ABS_MT_TOOL_TYPE,
-- 
1.7.11.rc3

